WooCommerce/Classes/Authentication/Navigation Exceptions/ApplicationPasswordDisabledViewModel.swift

```diff
@@ -34,14 +37,18 @@ struct ApplicationPasswordDisabledViewModel: ULErrorViewModel {
     let secondaryButtonTitle = Localization.secondaryButtonTitle
 
     func viewDidLoad(_ viewController: UIViewController?) {
-        // TODO: add tracks if necessary
     }
 
+    // Navigates back to the third last view controller in the stack if possible,
+    // otherwise it navigates to the root view controller.
     func didTapPrimaryButton(in viewController: UIViewController?) {
-        guard let viewController else {
-            return
+        guard let navigationController = viewController?.navigationController else { return }
+        let viewControllers = navigationController.viewControllers
+        if viewControllers.count >= 3 {
+            navigationController.popToViewController(viewControllers[viewControllers.count - 3], animated: true)
+        } else {
+            navigationController.popToRootViewController(animated: true)
         }
```

Comment:
The number 3 seems to be a magic number, it's not clear why we have to do this. Please consider my suggestion above about looking for the index of a specific view controller to pop to.

Reply:
Now I'm using the `previousVC` like in the comment above, please feel free to check changes here: 9b480ba48fc8c2f0c04aac77c43c512fb103c82e 🙏